src/rmind/components/objectives/policy.py

```diff
@@ -72,8 +72,13 @@ def forward(self, episode: Episode, encoder: Module) -> TensorDict:
             SpecialToken.OBSERVATION_SUMMARY,
         ))
 
+        waypoints = episode.input_embeddings[Modality.POINTS, "waypoints"][:, -1].mean(
```

Comment:
just to double check -- do we intentionally disregard encoder output and take waypoints from episode embeddings?

Reply:
good point, but yes, it's intentional. The idea is that encoder might keep all the necessary information in `OBSERVATION_SUMMARY` and on policy we provide "raw" waypoints. It makes even more sense if model is finetuned on different data with frozen encoder